rollup/da_syncer/batch_queue.go

```diff
@@ -59,50 +57,39 @@ func (bq *BatchQueue) NextBatch(ctx context.Context) (da.Entry, error) {
 }
 
 // getFinalizedBatch returns next finalized batch if there is available
-func (bq *BatchQueue) getFinalizedBatch() (da.Entry, bool) {
-	if len(bq.batches) == 0 {
-		return nil, false
-	}
-	var minBatchIndex uint64 = math.MaxUint64
-	for index := range bq.batches {
-		if index < minBatchIndex {
-			minBatchIndex = index
-		}
+func (bq *BatchQueue) getFinalizedBatch() da.Entry {
+	if bq.batches.Len() == 0 {
+		return nil
 	}
-	if minBatchIndex <= bq.lastFinalizedBatchIndex {
-		batch, ok := bq.batches[minBatchIndex]
 
-		// this won't happen because wew just found minBatchIndex among map keys, but need to leave this check to pass CI
-		if !ok {
-			return nil, false
-		}
-		bq.deleteBatch(minBatchIndex)
-		return batch, ok
+	batch := bq.batches.Peek()
+	if batch.BatchIndex() <= bq.lastFinalizedBatchIndex {
+		bq.deleteBatch(batch.BatchIndex())
+		return batch
 	} else {
-		return nil, false
+		return nil
 	}
 }
 
 // deleteBatch deletes data committed in the batch from map, because this batch is reverted or finalized
 // updates DASyncedL1BlockNumber
 func (bq *BatchQueue) deleteBatch(batchIndex uint64) {
-	batch, ok := bq.batches[batchIndex]
-	if !ok {
-		return
+	var batch da.Entry
+	for batch = bq.batches.Peek(); batch.BatchIndex() <= batchIndex; {
+		bq.batches.Pop()
+
+		if bq.batches.Len() == 0 {
+			break
+		}
+		batch = bq.batches.Peek()
 	}
```

Comment:
I think we shouldn't delete all batches that have a smaller index. In case when this method is called due to a revert batch event all smaller batches should still be stored